src/components/MyAccount/CheckoutsTab.tsx

```diff
@@ -0,0 +1,243 @@
+          />
+          <Text sx={{ marginBottom: "0px" }}> Renewal failed </Text>
+        </Box>
+      ),
+    }
+
+    useEffect(() => {
+      const lastDisabledTime = localStorage.getItem(
+        `lastDisabledTime-${checkout.id}`
+      )
+      if (lastDisabledTime) {
+        const twentyFourHoursAgo = new Date().getTime() - 8640000
+        if (parseInt(lastDisabledTime, 10) > twentyFourHoursAgo) {
+          setButtonDisabled(true)
+        }
+      }
+    }, [])
+
+    const handleClick = async () => {
+      const response = await fetch(
+        `${BASE_URL}/api/account/checkouts/renew/${checkout.id}`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({ patronId: patron.id }),
+        }
+      )
+      const responseData = await response.json()
+      if (responseData.message == "Renewed") {
+        setButtonDisabled(true)
+        setModalProps(successModalProps)
+        onOpen()
+        localStorage.setItem(
+          `lastDisabledTime-${checkout.id}`,
+          new Date().getTime().toString()
+        )
+      } else {
+        setModalProps(failureModalProps)
+        onOpen()
+        //TO-DO: Log error console.log("error", responseData)
+      }
+    }
+
+    return (
+      <>
+        <Button
+          sx={{
+            width: "-webkit-fill-available",
+          }}
+          buttonType="secondary"
+          id={`renew-${checkout.id}`}
+          onClick={handleClick}
+          isDisabled={isButtonDisabled}
+        >
+          Renew
+        </Button>
+        <Modal {...modalProps} />
+      </>
+    )
+  }
+
+  function getDueDate(date) {
+    const d = new Date(date)
+    const year = d.getFullYear()
+    const day = d.getDate()
+    const month = d.toLocaleString("default", { month: "long" })
+    return month + " " + day + ", " + year
+  }
+
+  function getTitle(checkout) {
+    const href = checkout.isResearch
+      ? `https://nypl.org/research/research-catalog/bib/b${checkout.bibId}`
+      : `https://nypl.na2.iiivega.com/search/card?recordId=${checkout.bibId}`
+    return checkout.isNyplOwned ? (
+      <Link href={href}>{checkout.title}</Link>
+    ) : (
+      <Text>{checkout.title}</Text>
+    )
+  }
+
+  const checkoutsData = checkouts.map((checkout) => [
+    getTitle(checkout),
+    checkout.barcode,
+    checkout.callNumber,
+    getDueDate(checkout.dueDate),
+    checkout.isResearch ? null : RenewButton(checkout, patron),
+  ])
+  return (
+    <>
+      {checkouts.length === 0 && (
+        <Box
+          sx={{
+            fontSize: "desktop.body.body1",
+            marginTop: "m",
+            marginBottom: "m",
+            display: "flex",
+            justifyContent: "flex-start",
+          }}
+        >
+          <span>You currently do not have any research items checked out.</span>
+        </Box>
+      )}
+      <Box
+        sx={{
+          fontSize: "desktop.body.body2",
```

Comment:
since these styles are going to be used for the requests tab as well, can they be exported to MyAccount.scss?